src/hooks/useAllPreviousData.ts

```diff
@@ -0,0 +1,13 @@
+import { useEffect, useRef } from "react";
+
+export const useAllPreviousData = <T>(data?: T[]): T[] => {
```

Comment:
**suggestion**: What about `usePrevious` hook

I'm wondering if we should have here something like [usePrevious](https://usehooks.com/usePrevious/) hook and lift data merging from `useEffect` to components?

Reply:
I've prepared two hooks one useAllPrevious to keep all previous data and the second for keeping only last prev data.